sway-fmt-v2/src/items/item_enum.rs

```diff
@@ -17,74 +17,167 @@ impl Format for ItemEnum {
         formatted_code: &mut FormattedCode,
         formatter: &mut Formatter,
     ) -> Result<(), FormatterError> {
-        let enum_variant_align_threshold = formatter.config.structures.enum_variant_align_threshold;
+        // Bring configurations into scope.
+        //
+        // Should small enums formatted into a single line.
+        let enum_lit_single_line = formatter.config.structures.small_structures_single_line;
 
-        if let Some(visibility_token) = &self.visibility {
-            write!(formatted_code, "{} ", visibility_token.span().as_str())?;
-        }
+        // Get the width limit of an enum to be formatted into single line if `enum_lit_single_line` is true.
+        let width_heuristics = formatter
+            .config
+            .heuristics
+            .heuristics_pref
+            .to_width_heuristics(&formatter.config.whitespace);
+        let enum_lit_width = width_heuristics.structure_lit_width;
+
+        let multiline = !enum_lit_single_line || self.get_formatted_len() > enum_lit_width;
+
+        format_enum(self, formatted_code, formatter, multiline)?;
+        Ok(())
+    }
+}
+
+/// Format the enum if the multiline is passed as false enum will be formatted into a single line.
+///
+/// Example (multiline : false):
+/// enum Foo { bar: u64,  baz: bool }
+///
+/// Example (multiline : true):
+/// enum Foo {
+///     bar: u64,
+///     baz: bool,
+/// }
```

Comment:
We should try to be weary of doc comment formatting - see my comment [here](https://github.com/FuelLabs/sway/pull/2058#discussion_r904440534).

Reply:
Ah yes, I did see that comment just wasn't on my mind. I'll fix that. 

I just did this one since I started to see the code get a little disorganized, shouldn't really impact the time frame too much. I'll stay focused on the MVP tho, just trying to stay organized and on top of things (:

Reply:
Also I agree with the above sentiment about alignment, it's something that was deliberated a bit before starting the new formatter